extractor/src/main/java/org/schabi/newpipe/extractor/services/soundcloud/extractors/SoundcloudCommentsInfoItemExtractor.java

```diff
@@ -77,6 +143,54 @@ public String getUrl() {
 
     @Override
     public String getThumbnailUrl() {
-        return json.getObject("user").getString("avatar_url");
+        return user.getString("avatar_url");
+    }
+
+    @Override
+    public Page getReplies() {
+        if (replyCount == CommentsInfoItem.UNKNOWN_REPLY_COUNT) {
+            final JsonArray replies = new JsonArray();
+            final CommentsInfoItemsCollector collector = new CommentsInfoItemsCollector(
+                    ServiceList.SoundCloud.getServiceId());
+            // SoundCloud has only comments and top level replies, but not nested replies.
+            // Therefore, replies cannot have further replies.
+            if (superComment == null) {
+                // Loop through all comments which come after the original comment
+                // to find its replies.
+                final JsonArray allItems = json.getArray(SoundcloudCommentsExtractor.COLLECTION);
+                boolean foundReply = false;
+                for (int i = index + 1; i < allItems.size(); i++) {
+                    final JsonObject comment = allItems.getObject(i);
+                    if (SoundcloudParsingHelper.isReplyTo(item, comment)) {
+                        replies.add(comment);
+                        collector.commit(new SoundcloudCommentsInfoItemExtractor(
+                                json, i, comment, url, item));
+                        foundReply = true;
+                    } else if (foundReply) {
+                        // Only the comments directly after the original comment
+                        // having the same timestamp are replies to the original comment.
+                        // The first comment not having the same timestamp
+                        // is the next top-level comment.
+                        break;
+                    }
+                }
+            }
+            replyCount = replies.size();
```

Comment:
Are you creating the `replies` array just to keep track of the count? Can't you just set `replyCount` to 0 at the beginning and also use it to replace the `replies` array and the `foundReply` boolean?

Reply:
Good catch. I used the `replies` List to retrieve other info previously. It's not needed anymore.